R/plot_combined_outputs_pdf.R

```diff
@@ -0,0 +1,123 @@
+#' @param location_data a `data.frame` containing information about the locations
+#'   being forecast. Assumed to contain a "location" column that may be joined with
+#'   that in the provided `model_out_tbl` and `target_ts`, plus a "location_name"
+#'   column of full location names (instead of abbreviations or fips codes). Any
+#'   other columns will be ignored. Defaults to NULL, in which case the values in
+#'   the "location" column will be used as plot labels.
+#' @param reference_date string of the reference date for the forecasts.
+#'   Must be in the ymd format, with yyyy-mm-dd format recommended.
+#' @param intervals numeric vector of prediction interval levels to plot for the
+#'   quantile output type. Provided levels may be 0.5, 0.8, 0.9, 0.95.
+#'   NULL means no interval levels are plotted. Defaults to c(.5, .8, .95).
+#' @param cats_ordered character vector ordering the pmf output type IDs, which
+#'   provides the order in which the categories are stacked in the output plot.
+#'   Defaults to NULL, in which case pmf categories are ordered alphabetically.
+#' @param quantile_title string providing a title for the plot of quantile forecasts.
+#'   Defaults to NULL, in which case no title is used.
+#' @param pmf_title string providing a title for the plot of pmf forecasts.
+#'   Defaults to NULL, in which case no title is used.
+#'
+#' @return NULL
+#'
+#' @export
+plot_combined_outputs_pdf <- function(model_out_tbl, target_ts,
+                                      location_data = NULL, reference_date,
+                                      intervals = c(0.5, 0.8, 0.95), cats_ordered = NULL,
+                                      quantile_title = NULL, pmf_title = NULL) {
+
+  if (!inherits(model_out_tbl, "model_out_tbl")) {
+    model_out_tbl <- hubUtils::as_model_out_tbl(model_out_tbl)
+  }
+
+  forecast_types <- unique(model_out_tbl$output_type)
+  if (!all(c("quantile", "pmf") %in% forecast_types)) {
+    cli::cli_abort("{.arg model_out_tbl} must contain both the quantile and pmf output type")
+  }
+
+  validate_target_ts(target_ts)
+
+  if (is.null(location_data)) {
+    location_col <- "location"
+  } else {
+    req_loc_col <- c("location", "location_name")
+    if (!all(req_loc_col %in% names(location_data))) {
+      cli::cli_abort("{.arg location_data} must contain the columns {.val {req_loc_col}}")
+    }
+    location_col <- "location_name"
+    model_out_tbl <- dplyr::left_join(model_out_tbl, location_data, by = "location")
+    target_ts <- dplyr::left_join(target_ts, location_data, by = "location")
+  }
+
+  if (length(reference_date) > 1) {
+    cli::cli_abort("Only one {.arg reference_date} may be provided")
+  } else {
+    reference_date <- validate_ymd_date(reference_date, arg_name = "reference_date")
+  }
+
+  cats_actual <- unique(model_out_tbl$output_type_id[model_out_tbl$output_type == "pmf"])
+  if ((!is.character(cats_ordered) && !is.null(cats_ordered)) || !all(cats_ordered %in% cats_actual)) {
```

Comment:
Do you want `setequal` here?
```suggestion
  if ((!is.character(cats_ordered) && !is.null(cats_ordered)) || !setequal(cats_ordered, cats_actual)) {
```